conjure/errorwriter.go

```diff
@@ -217,6 +264,130 @@ func astErrorCodeMethod(errorDefinition spec.ErrorDefinition, info types.PkgInfo
 	}
 }
 
+// astErrorCauseMethod generates Cause function for an error, for example:
+//
+//  func (e *MyNotFound) Cause() error {
+//  	return e.cause
+//  }
+func astErrorCauseMethod(errorDefinition spec.ErrorDefinition, info types.PkgInfo) astgen.ASTDecl {
+	return &decl.Method{
+		Function: decl.Function{
+			Name: "Cause",
+			FuncType: expression.FuncType{
+				ReturnTypes: []expression.Type{
+					expression.ErrorType,
+				},
+			},
+			Body: []astgen.ASTStmt{
+				statement.NewReturn(
+					expression.NewSelector(expression.VariableVal(errorReceiverName), causeField),
+				),
+			},
+			Comment: "Cause returns the underlying cause of the error, or nil if none.\nNote that cause is not serialized and sent over the wire.",
+		},
+		ReceiverName: errorReceiverName,
+		ReceiverType: expression.Type(errorDefinition.ErrorName.Name).Pointer(),
+	}
+}
+
+// astErrorStackTraceMethod generates StackTrace function for an error, for example:
+//
+//  func (e *MyNotFound) StackTrace() error {
```

Comment:
the return type here should have werror.Stacktrace

Reply:
Fixed.